pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -239,6 +239,17 @@ func (pr *PipelineRunStatus) MarkResourceNotConvertible(err *CannotConvertError)
 	})
 }
 
+// MarkResourceFailed sets the ConditionSucceeded condition to ConditionFalse
+// based on an error that occurred and a reason
+func (pr *PipelineRunStatus) MarkResourceFailed(reason string, err error) {
+	pipelineRunCondSet.Manage(pr).SetCondition(apis.Condition{
+		Type:    apis.ConditionSucceeded,
+		Status:  corev1.ConditionFalse,
+		Reason:  reason,
+		Message: err.Error(),
```

Comment:
i wonder if we want to pass the error message through as is in the message field; it might be easier for the user to pass in a more specific message about what went wrong? often when we write error messages i think we have developers in mind more than the users

Reply:
It was always pretty much the same in the original code, which is why I went this direction.
But it's ok to change it back to separated ones.

Reply:
This was not done otherwise in https://github.com/tektoncd/pipeline/pull/2749